subsys/net/lib/download_client/src/http_download_client.c

```diff
@@ -26,6 +26,29 @@ BUILD_ASSERT_MSG(CONFIG_NRF_DOWNLOAD_MAX_FRAGMENT_SIZE <=
 		 CONFIG_NRF_DOWNLOAD_MAX_RESPONSE_SIZE,
 	"The response buffer must accommodate for a full fragment");
 
+static int socket_timeout_set(int fd)
+{
+	int err;
+
+	if (CONFIG_NRF_DOWNLOAD_CLIENT_SOCK_TIMEOUT_MS == K_FOREVER) {
+		return 0;
+	}
+
+	struct timeval timeo = {
+		.tv_sec = CONFIG_NRF_DOWNLOAD_CLIENT_SOCK_TIMEOUT_MS,
+	};
+
+	LOG_INF("Configuring socket timeout (%ld ms)", timeo.tv_sec);
+
+	err = setsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &timeo, sizeof(timeo));
+	if (err) {
+		LOG_WRN("Failed to set socket timeout, errno %d", errno);
+		return -1;
```

Comment:
nit: You could return the actual error code here (`return -errno;`)

Reply:
Yeap, I agree

Reply:
Done!